### PR TITLE
Upgrade to webpack 3.x

### LINKED ISSUE #50: Update boilerplate to webpack 2
Why? because webpack 1 is depricated.

### PR DESCRIPTION
**Pull request by jesse-mm**
Future-proving seng-boilerplate, upgraded to Webpack-3.

- Upgraded all packages in the boilerplate.
- Removed unused packages or deprecated packages
- Remove remapping as karma-coverage-istanbul-reporter resolves remap problems
- Updated webpack configs (compatibility) mostly
- Updated karma config
- Updated webpack scripts for creating builds
- Fixed webpack dev-server for use with webpack 3.x

This ticket resolves #50

### CODE REVIEW

File: script/package-dist.js [hunk 1]
```diff
@@ -2,78 +2,39 @@
...
-	const archive = archiver('zip', {});
-	return createArchive(archive, to, distName + '.zip');
-};
+const createDistTar = () => createArchive(
+	archiver('tar', { gzip: true, gzipOptions: { level: 9 } }),
+	src,
+	`${dst}.tar.gz`
+);
 
-const createEs6Zip = () =>
```
Comment (ThaNarie) [inline script/package-dist.js]: Why is es6 removed? Or is the normal output now es6 by default?

Comment (jesse-mm) [reply script/package-dist.js]: Not really know why it was there I just followed the output of the current builds which contain a folder with ES6. We could however package the ES6 version in a seperate package.

Comment (ThaNarie) [reply script/package-dist.js]: It was there to be downloadable from an external location.
All other exports (amd, umd, commonjs, browser) are exported to a single file. However, es6 consists of multiple files, so when creating a single zip you can download that version.

Comment (jesse-mm) [reply script/package-dist.js]: Updated as discussed!

**Pull request merged**